File: api/app/Models/PromotionItem.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\SoftDeletes;

class PromotionItem extends Model
{
    use SoftDeletes;
    protected $fillable = ['promotion_id', 'product_type_id', 'product_id', 'quantity', 'unit_price', 'notes'];

    protected $casts = [
        'quantity'   => 'integer',
        'unit_price' => 'decimal:2',
    ];

    public function promotion(): BelongsTo
    {
        return $this->belongsTo(Promotion::class);
    }

    public function productType(): BelongsTo
    {
        return $this->belongsTo(ProductType::class);
    }

    public function product(): BelongsTo
    {
        return $this->belongsTo(Product::class);
    }
}

File: api-dotnet/AdminHiitop.Api/Domain/Sales/Entities/PromotionItem.cs
using AdminHiitop.Api.Domain.Catalog.Entities;
using AdminHiitop.Api.Domain.Common;
using AdminHiitop.Api.Domain.Inventory.Entities;

namespace AdminHiitop.Api.Domain.Sales.Entities;

public sealed class PromotionItem : AuditableEntity
{
    public int PromotionId { get; set; }
    public int? ProductTypeId { get; set; }
    public int? ProductId { get; set; }
    public int Quantity { get; set; }
    public decimal? UnitPrice { get; set; }
    public string? Notes { get; set; }

    public Promotion Promotion { get; set; } = null!;
    public ProductType? ProductType { get; set; }
    public Product? Product { get; set; }
}
